tools/polarshift.rb

```diff
@@ -497,28 +509,36 @@ def clone_run(test_run_id: nil, options: nil)
       template_hash = {}
       case_status = options.status
       run_title = options.title
+      # if no statuss is given then we clone all cases
       valid_statuses = ['Passed', 'Failed', 'Waiting', 'Running', 'Blocked']
       query_result = polarshift.get_run_smart(project, test_run_id, with_cases: 'full')
       if case_status
         case_status = case_status.downcase.capitalize
         raised "Invalid status given, only #{valid_statuses} are accepted" unless valid_statuses.include? case_status
         filtered_cases = query_result['records']["TestRecord"].select {|r| r['result'] == case_status}
-        # user wants further filter by specifying subteam as as a single or
-        # multiple subteam by putting them in a csv
         if opts[:subteam]
-          subteams = opts[:subteam].split(',')
-          target_cases = []
-          filtered_cases.each do |fc|
-            fc_subteam = get_subcomponent(fc)
-            target_cases << fc if subteams.include? fc_subteam
-          end
-          filtered_cases = target_cases
+          filtered_cases, subteams = filtered_cases_by_subteam(subteam_opt: opts[:subteam], cases: filtered_cases)
+          run_title ||= "Clone of '#{case_status}' #{subteams}  cases for #{test_run_id}"
         end
         tc_ids = filtered_cases.map {|c| c['test_case']['id']}
-        run_title ||= "Clone of '#{case_status}' cases for #{test_run_id}"
+        if tc_ids.count == 0
+          print "No cases found matching your criteria\n"
+          exit
+        end
+        run_title ||= "Clone of '#{case_status}'  cases for #{test_run_id}"
       else
-        tc_ids = extract_test_case_ids(query_result)
-        run_title ||= "Clone of #{test_run_id}"
+        filtered_cases = query_result['records']["TestRecord"]
+        if opts[:subteam]
+          filtered_cases, subteams = filtered_cases_by_subteam(subteam_opt: opts[:subteam], cases: filtered_cases)
+          run_title ||= "Clone of '#{case_status}' #{subteams}  cases for #{test_run_id}"
```

Comment:
We do not have `case_status` in the `else` block.

Reply:
@liangxia good catch, fixed now.